dockerfiles/mlrun/requirements.txt

```diff
@@ -4,3 +4,6 @@ scikit-learn~=1.0
 seaborn~=0.11.0
 scikit-plot~=0.3.7
 mpi4py~=3.1
+# sqlalchemy version should be aligned with mlrun-api due to common use with the model monitoring database
+sqlalchemy~=1.4
+pymysql~=1.0
```

Comment:
1. Why do you require `pymysql`? It was not installed before on the `mlrun` Docker image.
2. Isn't the right place to specify the constraint for `sqlalchemy` is on `./requirements.txt`?
   It comes from the `alembic` dependency:
   https://github.com/mlrun/mlrun/blob/6fb0e9912373d791ad7ce4b7adec3360759db7c0/requirements.txt#L36
   ```
   SQLAlchemy==2.0.20
   └── alembic==1.12.0 [requires: SQLAlchemy>=1.3.0]
   ```
   (The output of
   ```
   pipdeptree --reverse --packages sqlalchemy
   ```

Reply:
1. pymysql is used in the API to connect to the mysql DB. I assumed model monitoring is working similarly. If not, it can be removed.
2. About where to specify the requirements - alembic should not really be part of the mlrun client. I plan to move it to the API in https://github.com/mlrun/mlrun/pull/4213 but encountered another issue where we need sqlalchemy in datastore so it seems sqlalchemy must be a part of the mlrun client after all.
3. To allow sqlalchemy 2.0 in the client we need to ensure no discrepancies with the API which uses 1.4.
4. Good point, agree we should have test for this if possible.

Reply:
2. moved to mlrun client.
4. We don't have tests for model monitoring with mysql as it only works in the CE at the moment. Therefore until we have an organized mlrun/ce tests, it's our responsibility to validate that nothing breaks in the CE each time we are doing something new.

Reply:
> 2. About where to specify the requirements - alembic should not really be part of the mlrun client. I plan to move it to the API in [[Requirements] Move alembic to API #4213](https://github.com/mlrun/mlrun/pull/4213) but encountered another issue where we need sqlalchemy in datastore so it seems sqlalchemy must be a part of the mlrun client after all.

@alonmr regarding 2 - I'd argue that as long as we have these usages (just some examples):
1. https://github.com/mlrun/mlrun/blob/6fb0e9912373d791ad7ce4b7adec3360759db7c0/mlrun/common/db/sql_session.py#L18-L21
2. https://github.com/mlrun/mlrun/blob/6fb0e9912373d791ad7ce4b7adec3360759db7c0/mlrun/model_monitoring/stores/models/mysql.py#L17-L18
3. https://github.com/mlrun/mlrun/blob/6fb0e9912373d791ad7ce4b7adec3360759db7c0/mlrun/utils/db.py#L18

`sqlalchemy` should be specified as an explicit dependency of the `mlrun` package, or at least in the extras.

Reply:
Agree IMO it should be an extra requirement.